lib/file_store/middleware/prefix.ex

```diff
@@ -54,8 +54,8 @@ defmodule FileStore.Middleware.Prefix do
       FileStore.rename(store.__next__, put_prefix(src, store), put_prefix(dest, store))
     end
 
-    def upload(store, source, key) do
-      FileStore.upload(store.__next__, source, put_prefix(key, store))
+    def upload(store, source, key, opts) do
```

Comment:
Same thought here.

Reply:
It will break some what, but a compiler error on the down stream consumer will be returned. They will be notified that they need to fill the rest of it in.